src/screens/Events/EventsList.jsx

```diff
@@ -140,18 +142,63 @@ const EventsList = ({
     convertTime: convertTime,
   };
 
+  const [pageNum, setPageNum] = useState(1);
+  const onPageChange = () => {
+    setPageNum(pageNum + 10);
+  };
+
+  const formatDate = (date) => {
+    const split = date.split(" ");
+    return split[1] + " " + split[2] + ", " + split[3];
+  };
+
+  var pastDate = "";
+
+  const updateDate = (event) => {
+    const past = pastDate;
+    pastDate = formatDate(new Date(event.date).toDateString());
+    return past;
+  };
+
+  const past = () => {
+    return pastDate;
+  };
+
   if (!isHomePage) {
     return (
       <Styled.Container>
         {events.map((event) => (
-          <EventCard
-            key={event._id}
-            event={event}
-            user={user}
-            functions={functions}
-            onRegisterClicked={onRegisterClicked}
-          />
+          <>
+            {updateDate(event) !==
+            formatDate(new Date(event.date).toDateString()) ? (
+              <Text
+                type="subheader"
+                className="m\y-2"
+                text={formatDate(new Date(event.date).toDateString())}
+              />
+            ) : (
+              <></>
+            )}
+            <EventCard
+              key={event._id}
+              event={event}
+              user={user}
+              functions={functions}
+              onRegisterClicked={onRegisterClicked}
+              private={event.isPrivate}
+              dateDisplay={false}
+            />
+          </>
         ))}
+        {/* <div className="flex items-center justify-center text-center">
+          <Pagination
+            currentPage={pageNum}
+            layout="table"
+            onPageChange={onPageChange}
```

Comment:
Look at the existing pagination implementations for ideas. The idea is that we slice the list of events based on what page is selected